client/src/app/site/services/auth-check.service.ts

```diff
@@ -0,0 +1,120 @@
+        } else {
+            this.osRouter.navigateToLogin();
+            return false;
+        }
+    }
+
+    private hasPerms(basePerm: Permission | Permission[], omlPerm?: OML | OML[]): boolean {
+        let result = true;
+        if (!!basePerm && !!this.activeMeeting.meetingId) {
+            const toCheck = Array.isArray(basePerm) ? basePerm : [basePerm];
+            result = this.operator.hasPerms(...toCheck);
+        }
+        if (!omlPerm) {
+            return result;
+        }
+        const toCheck = Array.isArray(omlPerm) ? omlPerm : [omlPerm];
+        return result && this.operator.hasOrganizationPermissions(...toCheck);
+    }
+
+    private isMeetingSettingEnabled(meetingSetting?: keyof Settings): boolean {
+        if (!meetingSetting) {
+            return true;
+        }
+        return this.meetingSettingService.instant(meetingSetting) as boolean;
+    }
+
+    /**
+     * Handles a forbidden route. If the route is "/" (start page), It is tried to
+     * use a fallback route provided by AuthGuardFallbackRoutes. If this won't work
+     * or it wasn't the start page in the first place, the operator will be redirected
+     * to an error page.
+     */
+    public async handleForbiddenRoute(routeData: Data, segments: UrlSegment[]): Promise<void> {
+        if (segments.length === 0) {
+            // start page
+            const fallbackRoute = this.fallbackRoutesService.getFallbackRoute();
+            if (fallbackRoute) {
+                this.router.navigate([fallbackRoute]);
+                return;
+            }
+        }
+        const routeParams = await firstValueFrom(this.osRouter.currentParamMap);
+        const meetingId = routeParams?.[`meetingId`];
+
+        let routeDataArray = [];
+        if (routeData?.[`meetingPermissions`]) {
+            routeDataArray = routeDataArray.concat(routeData[`meetingPermissions`]);
+        }
+        if (routeData?.[`omlPermissions`]) {
+            routeDataArray = routeDataArray.concat(routeData[`omlPermissions`]);
+        }
+
+        if (meetingId) {
+            this.router.navigate([meetingId], {
+                queryParams: {
+                    meetingId,
+                    error: `Authentication Error`,
+                    msg: routeDataArray
+                }
+            });
+        } else {
+            this.router.navigate([`error`], {
+                queryParams: {
+                    error: `Authentication Error`,
+                    msg: routeDataArray
+                }
+            });
+        }
+    }
+
+    /**
+     * If the user requested a route without direct meaning, forward to their meaningful meeting
+     */
+    public forwardToOnlyMeeting(): void {
+        try {
+            const meetingId = this.operator.onlyMeeting;
+            this.router.navigate([meetingId]);
+        } catch (e) {
+            throw new Error(`Error when trying to forward to only meeting: ${e?.message}`);
+        }
+    }
```

Comment:
I would expect, that the `Auth**Check**Service` checks only if a user is, for example, authorized or authenticated (or both). These functions should be implemented at another place.